Initial buildchain setup

gdemonet opened the pull request:
This PR introduces a minimal Makefile to generate the Solution ISO from existing sources. It is responsible for building images, saving them into the expected layout, generating some files (mostly K8s manifests), and packaging them into a single ISO 9660 archive.

It also embeds `nicolast/static-container-registry` as a `git subrepo`, and includes the contents of `bitbucket.org/scality/scality-ui-template` (not as a `subrepo` for now, will change this in the future).

Please review by commit for your sanity.

### TODO

- [ ] include changes from [this PR](https://github.com/NicolasT/static-container-registry/pull/12) manually (will be merged into the subrepo when upstream is updated)
- [ ] fix Operator build outside of `GOPATH`

carlito-scality committed:
Integrate Template UI

This is a copy-paste of the scality/scality-ui-template repository,
until we merge it as a `git subrepo` (equivalent to a subtree).

Provides basic UI functionality to authenticate users against API
Server, shows how to use scality/core-ui components, and rely on theme
configuration.

gdemonet committed:
Add UI deployment manifests

This initial version of manifests for deploying the Solution UI in a
Kubernetes cluster should embody the best practices MetalK8s recommends.

Some parts are rendered from the buildchain, and the `make` targets will
require more flexibility to have those manifests compatible with any K8s
cluster.

gdemonet committed:
Merge "static-container-registry" as a git subrepo

git subrepo clone (merge) https://github.com/nicolast/static-container-registry

subrepo:
  subdir:   "static-container-registry"
  merged:   "c757492"
upstream:
  origin:   "https://github.com/nicolast/static-container-registry"
  branch:   "master"
  commit:   "c757492"
git-subrepo:
  version:  "0.4.0"
  origin:   "https://github.com/ingydotnet/git-subrepo"
  commit:   "5d6aba9"

gdemonet committed:
Set up Makefile-based buildchain

We create a rather simple Makefile to generate an ISO to conform with
the MetalK8s standards.

gdemonet committed:
[TODO] Update README with build instructions

We add the relevant information to build the Solution ISO as well as
parts of its contents through `make` invocations.

sayf-eddine-scality commented on static-container-registry/static-container-registry.py:
```diff
@@ -0,0 +1,226 @@
+LOGGER = logging.getLogger(__name__)
+
+
+CONSTANTS = '''
```
this duplicates these locations  (/v2, /v2/, @404_tag) on multiple solutions deployment and nginx will crash on it

gdemonet replied:
Thanks, will fix (there's an option to pass to the script IIRC)

gdemonet replied:
Oh actually not, need to fix upstream, I'll do a PR and include the change here in the meantime.

gdemonet committed:
Re-generate Operator with Go modules

Recent Operator SDK versions (>=0.9.x) allow to use Go modules instead
of `dep` for dependency management. This avoids vendoring the world in
our repository, and allows us to build the Operator image from outside
of `GOPATH/src`.

Note that the newly generated Operator also changed from cluster-scoped
to namespace-scoped.

NaasCraft committed:
fixup! [TODO] Update README with build instructions
